src/main/scala/utils/SqlUtils.scala

```diff
@@ -35,55 +50,81 @@ object SqlUtils:
+      case java.sql.Types.NUMERIC => Success(ArcaneType.BigDecimalType(jdbcTypeInfo.precision, jdbcTypeInfo.scale))
 
-    // numeric is functionally identical to decimal
-    // see: https://learn.microsoft.com/en-us/sql/t-sql/data-types/decimal-and-numeric-transact-sql
-    case java.sql.Types.NUMERIC => Success(ArcaneType.BigDecimalType(precision, scale))
+      // The SQL Server text and ntext types map to the JDBC LONGVARCHAR and LONGNVARCHAR type, respectively.
+      // see: https://learn.microsoft.com/en-us/sql/connect/jdbc/understanding-data-type-differences?view=sql-server-ver16#character-types
+      case java.sql.Types.LONGVARCHAR  => Success(ArcaneType.StringType)
+      case java.sql.Types.LONGNVARCHAR => Success(ArcaneType.StringType)
+      case java.sql.Types.DOUBLE       => Success(ArcaneType.DoubleType)
+      case java.sql.Types.INTEGER      => Success(ArcaneType.IntType)
 
-    // The SQL Server text and ntext types map to the JDBC LONGVARCHAR and LONGNVARCHAR type, respectively.
-    // see: https://learn.microsoft.com/en-us/sql/connect/jdbc/understanding-data-type-differences?view=sql-server-ver16#character-types
-    case java.sql.Types.LONGVARCHAR  => Success(ArcaneType.StringType)
-    case java.sql.Types.LONGNVARCHAR => Success(ArcaneType.StringType)
-    case java.sql.Types.DOUBLE       => Success(ArcaneType.DoubleType)
-    case java.sql.Types.INTEGER      => Success(ArcaneType.IntType)
+      case java.sql.Types.FLOAT => Success(ArcaneType.FloatType)
+      // The ISO synonym for real is float(24).
+      // See: https://learn.microsoft.com/en-us/sql/t-sql/data-types/float-and-real-transact-sql?view=sql-server-ver16
+      case java.sql.Types.REAL => Success(ArcaneType.FloatType)
 
-    case java.sql.Types.FLOAT => Success(ArcaneType.FloatType)
-    // The ISO synonym for real is float(24).
-    // See: https://learn.microsoft.com/en-us/sql/t-sql/data-types/float-and-real-transact-sql?view=sql-server-ver16
-    case java.sql.Types.REAL => Success(ArcaneType.FloatType)
+      case java.sql.Types.SMALLINT  => Success(ArcaneType.ShortType)
+      case java.sql.Types.TINYINT   => Success(ArcaneType.ShortType)
+      case java.sql.Types.TIME      => Success(ArcaneType.TimeType)
+      case java.sql.Types.NCHAR     => Success(ArcaneType.StringType)
+      case java.sql.Types.NVARCHAR  => Success(ArcaneType.StringType)
+      case java.sql.Types.VARCHAR   => Success(ArcaneType.StringType)
+      case java.sql.Types.VARBINARY => Success(ArcaneType.ByteArrayType)
+      case java.sql.Types.ARRAY =>
+        jdbcTypeInfo match {
+          case f: JdbcArrayTypeInfo =>
+            toArcaneType(f.arrayBaseElementType).map(elementType => ArcaneType.ListType(elementType, 0))
+          case _ =>
+            Failure(
+              new IllegalArgumentException(
+                s"Type ${jdbcTypeInfo.name} has java.sql.types.Array identifier, but is not provided as JdbcArrayTypeInfo"
```

Comment:
It's a column name, right?

```suggestion
                s"Type of the column ${jdbcTypeInfo.name} has java.sql.types.Array identifier, but is not provided as JdbcArrayTypeInfo"
```